src/pages/static/js/settings.js

```diff
@@ -0,0 +1,219 @@
+  if (!ipcRenderer) {
+    console.warn('IPC not available, using defaults');
+    loadDefaultSettings();
+    return;
+  }
+  
+  try {
+    console.log('Loading settings from backend...');
+    const settings = await ipcRenderer.invoke('settings-get-all');
+    console.log('Settings loaded:', settings);
+    populateFormFields(settings);
+  } catch (error) {
+    console.error('Failed to load settings:', error);
+    // Fallback to defaults if backend fails
+    loadDefaultSettings();
+  }
+}
+
+// Populate form fields with settings data
+function populateFormFields(settings) {
+  const searchEngine = document.getElementById('search-engine');
+  const themeToggle = document.getElementById('theme-toggle');
+  const showClock = document.getElementById('show-clock');
+  const wallpaperSelector = document.getElementById('wallpaper-selector');
+  
+  if (searchEngine && settings.searchEngine) {
+    searchEngine.value = settings.searchEngine;
+  }
+  if (themeToggle && settings.theme) {
+    themeToggle.value = settings.theme;
+  }
+  if (showClock && typeof settings.showClock === 'boolean') {
+    showClock.checked = settings.showClock;
+  }
+  if (wallpaperSelector && settings.wallpaper) {
+    wallpaperSelector.value = settings.wallpaper;
+  }
+  
+  console.log('Form fields populated with settings');
+}
+
+async function saveSettingToBackend(key, value) {
+  if (!ipcRenderer) {
+    console.warn('IPC not available, setting not saved:', key, value);
+    return;
+  }
+  
+  try {
+    console.log('Saving setting to backend:', key, value);
+    const result = await ipcRenderer.invoke('settings-set', key, value);
+    console.log('Setting saved successfully:', result);
+    
+    // TODO: Show success notification in future commits
+    return result;
+  } catch (error) {
+    console.error('Failed to save setting:', error);
+    alert(`Failed to save ${key}: ${error.message}`);
+    
+    // Reload settings to revert form to current backend state
+    await loadSettingsFromBackend();
+    throw error;
+  }
+}
+
+async function resetSettingsToDefaults() {
+  if (!ipcRenderer) {
+    console.warn('IPC not available, cannot reset settings');
+    return;
+  }
+  
+  try {
+    if (!confirm('Are you sure you want to reset all settings to defaults? This cannot be undone.')) {
+      return;
+    }
+    
+    console.log('Resetting settings to defaults...');
+    const result = await ipcRenderer.invoke('settings-reset');
+    console.log('Settings reset successfully:', result);
+    
+    // Reload form with new default values
+    await loadSettingsFromBackend();
+    
+    alert('Settings have been reset to defaults.');
+    return result;
+  } catch (error) {
+    console.error('Failed to reset settings:', error);
+    alert(`Failed to reset settings: ${error.message}`);
+    throw error;
+  }
+}
+
+// TODO: Add theme application functions
+function applyTheme(themeName) {
+  // TODO: Apply theme changes to current page
```

Comment:
For themes, I've added both light and dark combinations of colors here: https://github.com/p2plabsxyz/peersky-browser/blob/dedba45b1a6209bcbb56634b447cb8eafbaa00a7/src/pages/theme/vars.css#L11

We can update those values later based on how they look. We can have 6 themes; light, dark, violet, green, cyan, and yellow.

See the examples here: https://holo-themes.vercel.app/